platform-services.md

```diff
@@ -0,0 +1,215 @@
+platform’s view hierarchy. For example, on iOS, you add a `FlutterViewController` object to
+interact with a `UIView`.  Similarly, on Android, you add a `FlutterView` object to your app’s
+`android.view.View` hierarchy. Your app can then communicate between the Dart and iOS or
+Android-specific parts of the app by using a simple message-passing interface.
+
+### Message passing {#message-passing}
+
+Flutter offers a simple message-passing interface between your platform-specific code and the
+Dart code. The message passing interface is asynchronous, bi-directional, and supports optional
+reply routing. Messages consist of strings encoded in JSON format, and these is no facility for
+transferring handles to other objects along with messages. You are responsible for encoding and
+decoding messages sent through this mechanism.
+
+## Get the `hello_services` sample {#the-hello_services-sample}
+
+The Flutter SDK includes a reference sample project called
+[`hello_services`](https://github.com/flutter/flutter/tree/master/examples/hello_services)
+that demonstrates how to connect to the geolocation service within an iOS or Android application
+by using the `FlutterView` interface.
+
+The `hello_services` project comes with helper code to handle the passing of
+JSON-formatted messages to invoke platform services. If your app needs access to functionality
+provided by platform-specific iOS or Android services, we strongly encourage you to
+[download](https://github.com/flutter/flutter/tree/master/examples/hello_services) and use this
+sample project as the starter code to build your own Flutter app.
+
+The following diagram shows how the `hello_services` app uses the `FlutterView` interface to
+send and receive messages to platform and third-party services on iOS and Android.
+
+ ![hello_services architecture](/images/hello_services.png)
+
+**Fig 1.** Architecture of the `hello_services` sample.
+
+The Dart, iOS, and Android-specific components are related as follows:
+
+* In the Dart code, the sample app imports the
+[`flutter/services.dart`](http://docs.flutter.io/flutter/services/services-library.html)
+package. This package defines the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class,
+which provides a bridge to send and listen for JSON-encoded messages exchanged with a platform
+service on a <em>host</em> app (that is, the iOS or Android-specific parts of your app). The
+<code>hello_services</code> app uses the static methods of the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class
+in the [`State`](http://docs.flutter.io/flutter/material/State-class.html) object of a
+stateful widget to make services requests, then updates the UI accordingly.
+
+* On iOS, the sample app imports the Flutter framework library (`Flutter.h`) as a
+dependency. The library defines the <code>FlutterViewController</code> class, which is a
```

Comment:
The word "dependency" doesn't mean anything here.  Is there some other way to import things?  I'd strike the part of the sentence starting with "as a".